### PR TITLE
Feature/72 currency contract

### PR DESCRIPTION
**Pull request by sadiabbasi**

### CODE REVIEW

File: docs/guides/deploy.md [hunk 1]
```diff
@@ -75,6 +75,6 @@ If you want to have your applications sign blockchain transactions, see [Sign a
 To run the [Demo website](../../examples/#tonomy-demo-integration-application) in your network follow these extra steps:
```
Comment (theblockstalk) [inline docs/guides/deploy.md:75]: we also need to add instructions (before step 2) to deploy the oncoin.token contract

File: src/cli/bootstrap/bootstrap.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import deployContract from './deploy-contract';
 import path from 'path';
 import { createAntelopeAccount, createApp } from './create-account';
-import { EosioTokenContract, EosioUtil, setSettings } from '../../sdk/index';
+import { EosioTokenContract, EosioUtil, setSettings, OnoCoinContract } from '../../sdk/index';
```
Comment (theblockstalk) [inline src/cli/bootstrap/bootstrap.ts:4]: this class should be renamed to DemoTokenContract

File: src/cli/bootstrap/bootstrap.ts [hunk 3]
```diff
@@ -18,17 +19,28 @@ export default async function bootstrap(args: string[]) {
...
 
+        await createAntelopeAccount({ account: 'onocoin.tmy' }, signer);
+        await deployContract(
+            {
+                account: 'onocoin.tmy',
+                contractDir: path.join(__dirname, '../../Tonomy-Contracts/contracts/onocoin.tmy'),
+            },
+            signer
+        );
+        await onoCoinContract.create('1000000000 SYS', signer);
```
[the remark this answers is not in the document]
Comment (theblockstalk) [reply src/cli/bootstrap/bootstrap.ts:41]: `50000000000.0000 ONO` it should be, that will also set it to have 4 decimal places

File: src/sdk/services/blockchain/contracts/EosioTokenContract.ts [hunk 4]
```diff
@@ -3,6 +3,8 @@ import { API, Name, NameType } from '@wharfkit/antelope';
 import { Signer, transact } from '../eosio/transaction';
 import { getApi } from '../eosio/eosio';
 
+const CONTRACT_NAME = 'demo.tmy';
+
 class EosioTokenContract {
```
Comment (theblockstalk) [inline src/sdk/services/blockchain/contracts/EosioTokenContract.ts:8]: rename the class and file DemoTokenContract

File: src/sdk/services/blockchain/contracts/OnoCoinContract.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+                    quantity,
+                    memo: 'issued',
+                },
+            },
+        ];
+
+        return await transact(Name.from(CONTRACT_NAME), actions, signer);
+    }
+
+    async selfIssue(to: Name, quantity: string, signer: Signer): Promise<API.v1.PushTransactionResponse> {
```
Comment (theblockstalk) [inline src/sdk/services/blockchain/contracts/OnoCoinContract.ts:58]: this is not a function in the onocoin contract

File: src/sdk/services/blockchain/contracts/OnoCoinContract.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+                    quantity,
+                    memo: 'self issued',
+                },
+            },
+        ];
+
+        return await transact(Name.from(CONTRACT_NAME), actions, signer);
+    }
+
+    async addPerm(permission: NameType, signer: Signer) {
```
Comment (theblockstalk) [inline src/sdk/services/blockchain/contracts/OnoCoinContract.ts:80]: this is also not a function

File: src/sdk/services/blockchain/contracts/OnoCoinContract.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
+/* eslint-disable camelcase */
+import { API, Name, NameType } from '@wharfkit/antelope';
+import { Signer, transact } from '../eosio/transaction';
+import { getApi } from '../eosio/eosio';
+
+const CONTRACT_NAME = 'onocoin.tmy';
+
+class OnoCoinContract {
```
Comment (theblockstalk) [inline src/sdk/services/blockchain/contracts/OnoCoinContract.ts:8]: where is the transfer() function?

File: docs/guides/deploy.md [hunk 8]
```diff
@@ -46,25 +46,26 @@ Then you will need to deploy each service, with configuration so that they conne
...
     - For maximum scaleability, run using bare-metal servers.
     - Please <a href="https://tonomy.io/contact" target="_blank">contact us</a> for assistance running a production Antelope network or setting up governance.
-
-2. Create the `id.tmy` account on the blockchain
-3. Deploy the <a href="https://github.com/Tonomy-Foundation/Tonomy-Contracts/tree/master/contracts/id.tmy" target="_blank">id.tmy</a> contract to the `id.tmy` account on the blockchain
-4. [Register your applications](../../start/register-app) that you wish to connect to Tonomy ID
-5. Configure the software with the `config.json` file in the repository so that they are connected correctly using your domains, and to white-label the applications:
+2. Create the `onocoin.tmy` account on the blockchain
+3. Deploy the <a href="https://github.com/Tonomy-Foundation/Tonomy-Contracts/tree/master/contracts/onocoin.tmy" target="_blank">onocoin.tmy</a> contract to the `onocoin.tmy` account on the blockchain
+4. Create the `id.tmy` account on the blockchain
```
Comment (theblockstalk) [inline docs/guides/deploy.md:51]: after step 3, need to create and issue the new currency from the onocoin.tmy token contract

File: src/cli/bootstrap/bootstrap.ts [hunk 9]
```diff
@@ -18,16 +19,27 @@ export default async function bootstrap(args: string[]) {
...
+        await createAntelopeAccount({ account: 'onocoin.tmy' }, signer);
+        await deployContract(
+            {
+                account: 'onocoin.tmy',
+                contractDir: path.join(__dirname, '../../Tonomy-Contracts/contracts/onocoin.tmy'),
+            },
+            signer
+        );
+        await onoCoinContract.create('50000000000.0000 ONO', signer);
+        await onoCoinContract.issue('32500000000.0000 ONO', signer);
```
Comment (theblockstalk) [inline src/cli/bootstrap/bootstrap.ts:42]: issue function takes three arguments, create function takes two arguments
https://github.com/Tonomy-Foundation/Tonomy-Contracts/pull/76/files#diff-f2316c881043fc8b491b5a01cef0d3b3b893d8d8b32d68b62e3dfc7544305092R41

File: src/cli/bootstrap/bootstrap.ts [hunk 10]
```diff
@@ -18,16 +19,27 @@ export default async function bootstrap(args: string[]) {
...
 
         await createAntelopeAccount({ account: 'id.tmy' }, signer);
```
Comment (theblockstalk) [inline src/cli/bootstrap/bootstrap.ts:44]: after all the accounts are created, allocate the coins to them as per slide 23 in the deck https://www.canva.com/design/DAFzI6RvDKA/9RM1pllCtFvFTnbsbSY6XA/edit

account | allocation (% of total supply)
team.tm 15%
ecosystm.tmy 30%
private1.tmy 2.5%
private2.tmy 2.5%
private3.tmy 2.5%
public1.tmy 2.5%
public2.tmy 2.5%
public3.tmy 2.5%
operation.tmy (rename this) 40%

File: src/sdk/services/blockchain/contracts/OnoCoinContract.ts [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+                    quantity,
+                    memo: 'issued',
+                },
+            },
+        ];
+
+        return await transact(Name.from(CONTRACT_NAME), actions, signer);
+    }
+
+    async transfer(to: NameType, quantity: string, signer: Signer): Promise<API.v1.PushTransactionResponse> {
```
Comment (theblockstalk) [inline src/sdk/services/blockchain/contracts/OnoCoinContract.ts:58]: transfer has 4 arguments https://github.com/Tonomy-Foundation/Tonomy-Contracts/pull/76/files#diff-f2316c881043fc8b491b5a01cef0d3b3b893d8d8b32d68b62e3dfc7544305092R73

File: src/cli/bootstrap/bootstrap.ts (hunk 10 above)
Comment (sadiabbasi) [reply src/cli/bootstrap/bootstrap.ts:44]: if i change the opration.tmy to operation.tmy it throws maximum character exceed error.

Comment (theblockstalk) [reply src/cli/bootstrap/bootstrap.ts:44]: ah ok. then lets call it `ops.tmy`

**Review by theblockstalk — APPROVED** [review]

**Pull request merged**